.github/workflows/run-pre-commit.yaml

```diff
@@ -0,0 +1,23 @@
+name: Run pre-commit
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install pre-commit
+        run: |
+          pip install pre-commit
+  
+      - name: Run pre-commit
+        run: pre-commit run --all
```

Comment:
Is it possible to run individual `pre-commit` hooks at each step to make it easy to debug? Right now, I don't know where does a `pre-commit` fail, hence making the process of finding issues difficult.

Reply:
I don't mind the way it is now, but I see what you mean. So I don't mind adding extra steps as well, for a better layout.

Reply:
Thanks for the suggestion @HarshCasper. Having it all in one command will still output all of the failures.